### DISCUSSION TITLE
Tiny11 Azure AD join troubleshooting

### DISCUSSION THREAD

**Discussion by jbanes-edgeworthsecurity** [ntdevlabs/tiny11builder#248] [General]
Hello, testing out Tiny11 and I cannot  join it to an Azure AD. Have done a few days of troubleshooting, google-ing and chatgtp-ing. Is there some component I need to add back in to be able to join it to work or school Azure AD? Thank you

**Comment by Miiraak** [comment]
Hello!

#### If you used 'tiny11CoreMaker.ps1', the issue is likely related to the removal of critical components:
- **WinSxS removal** could be the main issue, as it prevents the restoration of essential features necessary for Azure AD connectivity.
- If your school or company requires **Windows Defender** to remain active, removing or disabling it may cause problems.
- Additionally, the **Windows Update mechanism** might be unavailable, making it impossible to apply necessary updates or security patches. This could result in failure to meet Azure AD requirements.

#### If you used 'tiny11maker.ps1', 
The situation is different. In this case, the removed applications are mainly productivity tools, entertainment apps, or non-essential features. These usually don’t interfere with Azure AD enrollment.

#### General considerations:
- If your organization uses **advanced security features** in Azure AD, such as Windows Hello for Business, ensure that **TPM** (Trusted Platform Module) is enabled and properly configured. TPM is crucial for these features and other security-related processes.
- To enroll in Azure AD, your device must be running **Windows Professional (Pro), Enterprise, or Education**. If you're using Tiny11 based on Windows Home, you won't be able to join an ENTRA ID.
- **OneDrive** and **Edge** are not direct prerequisites, but certain Azure AD features may depend on the services and protocols they facilitate. Reinstalling them might resolve some connectivity issues.

#### User troubleshooting tips:
- Run the command `dsregcmd /status` in PowerShell to check if your device is configured correctly for Azure AD. Ensure **AzureAdJoined** is listed as "YES".
- Verify that necessary services, like **Microsoft Account Sign-In Assistant (wlidsvc)**, are running. You can manage this through `services.msc`.
- Check your system’s edition by typing `winver` in the search bar-ensure it’s Pro, Enterprise, or Education.
- Confirm network connectivity by checking if **login.microsoftonline.com** resolves correctly using `nslookup`.

I hope this helps you, I’m not a pro, juste curious... forgive me if there are some misinformation...
I gathered some things and tried few ones but we are missing data from your configuration or the EntraID's one🐍
